Source/WTF/wtf/SortedArrayMap.h

```diff
@@ -159,7 +159,7 @@ template<ASCIISubset subset, typename CharacterType> constexpr std::make_unsigne
 template<ASCIISubset subset> template<unsigned size> constexpr ComparableASCIISubsetLiteral<subset>::ComparableASCIISubsetLiteral(const char (&characters)[size])
     : literal { ASCIILiteral::fromLiteralUnsafe(characters) }
 {
-    ASSERT_UNDER_CONSTEXPR_CONTEXT(allOfConstExpr(&characters[0], &characters[size - 1], [] (char character) {
+    ASSERT_UNDER_CONSTEXPR_CONTEXT(std::all_of(&characters[0], &characters[size - 1], [] (char character) {
```

Comment:
Might be able to use std::ranges::all_of(characters, [] …) to make this a bit shorter (can’t recall off hand if it works with that construction).

Reply:
(Generally, the ranges versions can probably be used in the PR).

Reply:
I was going to say `std::ranges::all_of` is missing from pre-Xcode 14.3, but turns out our oldest target is building against Xcode 14.3, so we're golden 😄

Reply:
I actually don't think I can do this in these asserts it turns out, since `std::range::all_of` returns a `constexpr bool` and the ranges version fails with `non-constexpr function 'WTFReportAssertionFailure' cannot be used in a constant expression`. The regular `std::all_of` returns just a regular `bool` and compiles fine in the assert (this is also true for `is_sorted`).

That said, if we have a preference for the `std::ranges` syntax since it looks cleaner, there's several other places where it could be swapped in and the `constexpr` difference shouldn't matter.